### PR TITLE
[REVIEW] Replace CNMeM with pool_memory_resource

### PR DESCRIPTION
**Pull request by harrism**
Replaces usage of `cnmem_memory_resource` in benchmark and test fixtures with `pool_memory_resource`.

Also adds option to test with `binning_memory_resource`.

Also replaces usage of deprecated `default_memory_resource` APIs with `current_device_resource` APIs in fixtures and JNI bindings.

### CODE REVIEW

File: cpp/tests/utilities/base_fixture.hpp [hunk 1]
```diff
@@ -219,6 +221,24 @@ class TempDirTestEnvironment : public ::testing::Environment {
...
+inline auto make_pool()
+{
+  return rmm::mr::make_owning_wrapper<rmm::mr::pool_memory_resource>(make_cuda());
+}
+
+inline auto make_binning()
+{
+  auto pool = make_pool();
+  auto mr   = rmm::mr::make_owning_wrapper<rmm::mr::binning_memory_resource>(pool);
+  for (std::size_t i = 18; i <= 22; i++) { mr->wrapped().add_bin(1 << i); }
```
Comment (davidwendt) [inline cpp/tests/utilities/base_fixture.hpp:238]: Could you add a comment on what is going on here?

Comment (harrism) [reply cpp/tests/utilities/base_fixture.hpp:240]: Done.

**Comment by harrism** [comment]
CI will pass once rapidsai/rmm#457 is merged.

File: cpp/tests/utilities/base_fixture.hpp (hunk 1 above)
Comment (harrism) [reply cpp/tests/utilities/base_fixture.hpp]: Also moved this into a constructor overload for `binning_memory_resource`.

**Pull request merged**